homeassistant/components/flux_led/__init__.py

```diff
@@ -1 +1,134 @@
-"""The flux_led component."""
+"""The Flux LED/MagicLight integration."""
+import asyncio
+from datetime import timedelta
+import logging
+
+from flux_led import BulbScanner
+import voluptuous as vol
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_HOST, CONF_NAME, CONF_TYPE
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryNotReady, PlatformNotReady
+from homeassistant.helpers.dispatcher import async_dispatcher_send
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator
+
+from .const import (
+    BULB_COORDINATOR,
+    DEFAULT_NETWORK_SCAN_INTERVAL,
+    DOMAIN,
+    SIGNAL_ADD_DEVICE,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+CONFIG_SCHEMA = vol.Schema({DOMAIN: vol.Schema({})}, extra=vol.ALLOW_EXTRA)
+
+PLATFORMS = ["light"]
+
+
+async def async_setup(hass: HomeAssistant, config: dict):
+    """Set up the Flux LED/MagicLight component."""
+    hass.data.setdefault(DOMAIN, {})
+
+    return True
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry):
+    """Set up Flux LED/MagicLight from a config entry."""
+
+    conf = entry.data
+
+    if conf[CONF_TYPE] == "auto":
+        scanner_api = BulbScanner()
+        if len(scanner_api.scan()) == 0:
+            raise PlatformNotReady("No FluxLED/MagicHome Bulbs found in network.")
+
+        coordinator = FluxLEDListUpdateCoordinator(
+            hass=hass,
+            name="flux_led_scanner",
+            update_interval=DEFAULT_NETWORK_SCAN_INTERVAL,
+        )
+
+        await coordinator.async_refresh()
+
+        if not coordinator.last_update_success:
+            raise ConfigEntryNotReady
+
+        hass.data[DOMAIN][entry.entry_id] = {
+            CONF_TYPE: "auto",
+            BULB_COORDINATOR: coordinator,
+        }
+
+    else:
+        hass.data[DOMAIN][entry.entry_id] = {
+            CONF_TYPE: "manual",
+            CONF_NAME: conf[CONF_NAME],
+            CONF_HOST: conf[CONF_HOST],
+            BULB_COORDINATOR: None,
+        }
+
+    for component in PLATFORMS:
+        hass.async_create_task(
+            hass.config_entries.async_forward_entry_setup(entry, component)
+        )
+
+    return True
+
+
+async def async_unload_entry(hass: HomeAssistant, entry: ConfigEntry):
+    """Unload a config entry."""
+    unload_ok = all(
+        await asyncio.gather(
+            *[
+                hass.config_entries.async_forward_entry_unload(entry, component)
+                for component in PLATFORMS
+            ]
+        )
+    )
+    if unload_ok:
+        hass.data[DOMAIN].pop(entry.entry_id)
+
+    return unload_ok
+
+
+class FluxLEDListUpdateCoordinator(DataUpdateCoordinator):
```

Comment:
Can we save the coordinator refactor for a follow up PR? Just do the config flow in this PR, would be best.

Reply:
It is actually the coordinator updates that fix the issues with the newer devices. Additionally the old YAML configuration included more complex device level configuration (custom effects etc) which would break if I just add the config flow to existing integration.

Reply:
Then we should use a separate target branch so that all PRs get released together.